### ISSUE TITLE
Default test does not pass with tox.

### ISSUE THREAD

**Issue by lslebodn** [release-engineering/pushsource#322]
#### Summary
Default test does not pass with tox

#### Issue Type
- Bug Report

#### Steps to reproduce
* clone master
* tox  # let `tox` run default test from envlist

#### Actual results
```
sh$ tox

//snip

Collecting virtualenv-api==2.1.18
  Using cached virtualenv_api-2.1.18-py2.py3-none-any.whl (9.5 kB)
Collecting wrapt==1.15.0
  Using cached wrapt-1.15.0-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (81 kB)
Collecting zipp==3.15.0
  Using cached zipp-3.15.0-py3-none-any.whl (6.8 kB)
Collecting pytz>=2015.7
ERROR: In --require-hashes mode, all requirements must have their versions pinned with ==. These do not:
    pytz>=2015.7 from https://files.pythonhosted.org/packages/2e/09/fbd3c46dce130958ee8e0090f910f1fe39e502cc5ba0aadca1e8a2b932e5/pytz-2022.7.1-py2.py3-none-any.whl#sha256=78f4f37d8198e0627c5f1143240bb0206b8691d8d7ac6d78fee88b78733f8c4a (from babel==2.12.1->-r test-requirements.txt (line 27))
WARNING: You are using pip version 21.3.1; however, version 23.0.1 is available.
You should consider upgrading via the '/home/user/projects/pushsource/.tox/py38/bin/python -m pip install --upgrade pip' command.

=============================================== log end ================================================
ERROR: could not install deps [-rtest-requirements.txt]; v = InvocationError('/home/user/projects/pushsource/.tox/py38/bin/python -m pip install -rtest-requirements.txt', 1)
_______________________________________________ summary ________________________________________________
ERROR:   py38: could not install deps [-rtest-requirements.txt]; v = InvocationError('/home/user/projects/pushsource/.tox/py38/bin/python -m pip install -rtest-requirements.txt', 1)

```

#### Expected results
Test passed without any failure with git master

#### Additional Information
There are more ways how to solve it.
A) run pip-compile with older version of python
* https://github.com/release-engineering/pushsource/blob/e0f1dfda17a18d7502cfa373b13172d32e4a7ca8/tox.ini#L36

BTW it does not work with 3.6
```
pip-compile run-test: commands[1] | pip-compile -U --generate-hashes requirements.in test-requirements.in -o test-requirements.txt
Could not find a version that matches importlib-resources<5.3,>=5.4 (from -r test-requirements.in (line 22))
Tried: 0.1.0, 0.1.0, 0.2, 0.2, 0.3, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.8, 1.0, 1.0.1, 1.0.1, 1.0.2, 1.0.2, 1.1.0, 1.1.0, 1.2.0, 1.2.0, 1.3.0, 1.3.0, 1.3.1, 1.3.1, 1.4.0, 1.4.0, 1.5.0, 1.5.0, 2.0.0, 2.0.0, 2.0.1, 2.0.1, 3.0.0, 3.0.0, 3.1.0, 3.1.0, 3.1.1, 3.1.1, 3.2.0, 3.2.0, 3.2.1, 3.2.1, 3.3.0, 3.3.0, 3.3.1, 3.3.1, 4.0.0, 4.0.0, 4.1.0, 4.1.0, 4.1.1, 4.1.1, 5.0.0, 5.0.0, 5.0.2, 5.0.2, 5.0.3, 5.0.3, 5.0.4, 5.0.4, 5.0.5, 5.0.5, 5.0.6, 5.0.6, 5.0.7, 5.0.7, 5.1.0, 5.1.0, 5.1.1, 5.1.1, 5.1.2, 5.1.2, 5.1.3, 5.1.3, 5.1.4, 5.1.4, 5.2.0, 5.2.0, 5.2.1, 5.2.1, 5.2.2, 5.2.2, 5.2.3, 5.2.3, 5.3.0, 5.3.0, 5.4.0, 5.4.0
There are incompatible versions in the resolved dependencies:
  importlib-resources (from -r test-requirements.in (line 22))
  importlib-resources>=5.4 (from virtualenv==20.17.1->pre-commit==2.17.0->-r test-requirements.in (line 10))
  importlib-resources<5.3 (from pre-commit==2.17.0->-r test-requirements.in (line 10))
ERROR: InvocationError for command /home/user/projects/pushsource/.tox/pip-compile/bin/pip-compile -U --generate-hashes requirements.in test-requirements.in -o test-requirements.txt (exited with code 2)
_______________________________________________ summary ________________________________________________
ERROR:   pip-compile: commands failed
```

B) change default python version in envlist in tox.ini (it works with 3.9)
* https://github.com/release-engineering/pushsource/blob/e0f1dfda17a18d7502cfa373b13172d32e4a7ca8/tox.ini#L2

And ideally upgrade requirements for python in setup.py
https://github.com/release-engineering/pushsource/blob/e0f1dfda17a18d7502cfa373b13172d32e4a7ca8/setup.py#L41

C) do not used pinned versions for test (but that would complicate "reproducible tests" locally and in github actions)
* https://github.com/release-engineering/pushsource/blob/e0f1dfda17a18d7502cfa373b13172d32e4a7ca8/tox.ini#L6

And maybe even some other

**Comment by lslebodn** [comment]
```
sh$ git log -1 --oneline 
e0f1dfd (HEAD -> master, origin/master, origin/HEAD) Merge pull request #318 from release-engineering/deps/pip-compile

sh$ tox -l
py38
static
docs
```

**Comment by lslebodn** [comment]
Closing, it was already fixed in #321
